simulation/executor/mock_tendermint.go

```diff
@@ -193,13 +193,17 @@ func randomDoubleSignEvidence(r *rand.Rand, params Params,
 	pastVoteInfos [][]abci.VoteInfo,
 	event func(route, op, evResult string), header tmproto.Header, voteInfos []abci.VoteInfo) []abci.Evidence {
 	evidence := []abci.Evidence{}
-	// return if no past times
-	if len(pastTimes) == 0 {
+	// return if no past times or if only 10 validators remaining in the active set
+	if len(pastTimes) == 0 || len(voteInfos) <= 10 {
 		return evidence
 	}
-
+	var n float64 = 1
 	// TODO: Change this to be markov based & clean this up
-	for r.Float64() < params.EvidenceFraction() {
+	// Right now we incrementally lower the evidence fraction to make
+	// it less likely to jail many validators in one run
+	// We should also add some method of including new validators into the set
+	// instead of being stuck with the ones we start with during initialization
+	for r.Float64() < (params.EvidenceFraction() / n) {
```

Comment:
Why don't we just say that if num validators remaining = 1, continue? (Don't test double signing the last validator?)

Reply:
Sounds good, just implemented that! Still needed to keep the counter for the following reasons:

1. The validator length doesn't update after appending to the evidence slice, so this leads to edge cases of still jailing too many
2. Dividing the evidence fraction by the iteration number sizeably slows down the jailing